pulser-core/pulser/_hamiltonian_data/hamiltonian_data.py

```diff
@@ -43,6 +45,22 @@
 )
 from pulser.sequence import Sequence
 
+
+class TrajectoryWithReps(NamedTuple):
+    """A NoiseTrajectory and the number of times it should be simulated."""
+
+    trajectory: NoiseTrajectory
+    reps: int
+
+
+class SamplesWithReps(NamedTuple):
+    """A trajectory, samples and how often they should be simulated."""
+
+    trajectory: NoiseTrajectory
+    samples: SequenceSamples
+    reps: int
```

Comment:
Thanks for adding these!
nit: I think it would improve readability to use the named entries where you can in the code (e.g. `x.trajectory` instead of `x[0]`)